tests/notebooks/kserve-integration.ipynb

```diff
@@ -0,0 +1,369 @@
+   "source": [
+    "### Submit Inference Service"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "f02a6871-da47-4f55-b7d3-80da300488f1",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'apiVersion': 'serving.kserve.io/v1beta1',\n",
+       " 'kind': 'InferenceService',\n",
+       " 'metadata': {'annotations': {'sidecar.istio.io/inject': 'false'},\n",
+       "  'creationTimestamp': '2023-08-07T09:50:16Z',\n",
+       "  'generation': 1,\n",
+       "  'managedFields': [{'apiVersion': 'serving.kserve.io/v1beta1',\n",
+       "    'fieldsType': 'FieldsV1',\n",
+       "    'fieldsV1': {'f:metadata': {'f:annotations': {'.': {},\n",
+       "       'f:sidecar.istio.io/inject': {}}},\n",
+       "     'f:spec': {'.': {},\n",
+       "      'f:predictor': {'.': {},\n",
+       "       'f:sklearn': {'.': {}, 'f:name': {}, 'f:storageUri': {}}}}},\n",
+       "    'manager': 'OpenAPI-Generator',\n",
+       "    'operation': 'Update',\n",
+       "    'time': '2023-08-07T09:50:14Z'}],\n",
+       "  'name': 'sklearn-iris',\n",
+       "  'namespace': 'test',\n",
+       "  'resourceVersion': '56838',\n",
+       "  'uid': 'ece28eaa-802a-45ce-a71d-22af323a3528'},\n",
+       " 'spec': {'predictor': {'model': {'modelFormat': {'name': 'sklearn'},\n",
+       "    'name': '',\n",
+       "    'resources': {},\n",
+       "    'storageUri': 'gs://kfserving-examples/models/sklearn/1.0/model'}}}}"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "client.create(isvc)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "e8cb648e-bf9d-455e-a40e-022c35772aa3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@retry(\n",
+    "    wait=wait_exponential(multiplier=2, min=1, max=10),\n",
+    "    stop=stop_after_attempt(30),\n",
+    "    reraise=True,\n",
+    ")\n",
+    "def assert_isvc_created(client, isvc_name):\n",
+    "    \"\"\"Wait for the Inference Service to be created successfully.\"\"\"\n",
+    "    assert client.is_isvc_ready(ISVC_NAME), f\"Failed to create Inference Service {isvc_name}.\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "50871dc2-9ef1-4458-813b-2d9dca03c6a1",
+   "metadata": {
+    "tags": [
+     "raises-exception"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "assert_isvc_created(client, ISVC_NAME)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1c03aeca-0828-4db1-9051-88eb25a26277",
+   "metadata": {},
+   "source": [
+    "## Perform Inference"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "8522c4e9-07b7-4bff-9b49-3675ff19bacc",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Inference URL: http://sklearn-iris.test.svc.cluster.local/v1/models/sklearn-iris:predict\n"
```

Comment:
I believe the inference URL will be different depending on the `domain-name` configuration, so it's not always going to be `svc.cluster.local`. I wonder if a note here would prevent confusion. Up to you.

Reply:
Indeed, currently the notebooks generally lack comments on what it is that we're seeing. I could add this comment, but it feels too specific given that I don't explain anything else. I mean, these are hardly intended to serve as tutorials.

Reply:
Yeah, I agree they are not tutorials, but it's good to have some information for the future in case we run into certain issues.
I was wondering if we should add a README.md file where we'd explain how charms should be deployed and configured (that's out of the scope of this PR) to run the UATs and keep each run exactly the same always.
We don't need to block the PR with this comment, but to keep in mind.